test/docfx.Test/e2e/E2ETest.cs

```diff
@@ -125,6 +133,60 @@ private static async Task<(string docsetPath, E2ESpec spec)> CreateDocset(string
             return (docsetPath, spec);
         }
 
+        private static void AddExtendConfig(string docsetPath,  string[] suppressedRules)
+        {
+            var docfxConfigPath = Path.Combine(docsetPath, "docfx.yml");
+            if (File.Exists(docfxConfigPath))
+            {
+                string configContent = File.ReadAllText(docfxConfigPath);
+                if (string.IsNullOrWhiteSpace(configContent))
+                {
+                    GenerateSuppressedRulesFile(docfxConfigPath, suppressedRules);
+                    return;
+                }
+
+                var match = ExtendConfigRegex.Match(configContent);
+                if (match.Success)
+                {
+                    var orignalExtendValue = match.Groups[1].ToString().TrimStart(' ');
+                    var content = ExtendConfigRegex.Replace(configContent,
+                        $"extend:{Environment.NewLine}- {s_extendFileName}{(orignalExtendValue != string.Empty ? Environment.NewLine + "- " + orignalExtendValue : string.Empty)}");
+                    File.WriteAllText(docfxConfigPath, content);
+                }
+                else
+                {
+                    if (!configContent.EndsWith("\n"))
+                    {
+                        File.AppendAllText(docfxConfigPath, Environment.NewLine);
+                    }
+                    File.AppendAllText(docfxConfigPath, $"extend: {s_extendFileName}");
+                }
+
+                var extendFilePath = Path.Combine(docsetPath, s_extendFileName);
+                GenerateSuppressedRulesFile(extendFilePath, suppressedRules);
+            }
+            else
+            {
+                GenerateSuppressedRulesFile(docfxConfigPath, suppressedRules);
+            }
+        }
+
+        private static void GenerateSuppressedRulesFile(string filePath, string[] suppressedRules)
```

Comment:
we may need 2 more PRs before this PR:

1. implement `--config` cmd args
2. auto exclude extend config from sdp

I'll work on 2, @928PJY can you work on 1?